pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -922,10 +922,19 @@ func (v dockerVersion) String() string {
 	return string(v)
 }
 
+func pruneSemver(semver string) string {
+	// prune strings like 1.10.4-28.gitf476348.fc23 to 1.10.4
+	parts := strings.SplitN(semver, "-", 2)
+	return parts[0]
+}
+
 func (v dockerVersion) Compare(other string) (int, error) {
-	if dockerversion.LessThan(string(v), other) {
+	v1 := pruneSemver(string(v))
```

Comment:
Kubelet uses the `String` function to get runtime version and send to apiserver as node status, it would be better to make `String()` return full version instead of pruned one. :)